app/controllers/Checkout.scala

```diff
@@ -46,18 +47,30 @@ object Checkout extends Controller with LazyLogging {
           yield BadRequest(views.html.checkout.payment(formWithErrors, userIsSignedIn = idUserOpt.isDefined))
       },
       userData => {
-        CheckoutService.processSubscription(userData, AuthenticationService.authenticatedUserFor(request)).map {
-          case Some(memberId) =>
-            Redirect(routes.Checkout.thankyou())
-          case None =>
-            Redirect(routes.Checkout.renderCheckout())
-        }
+        val idUserOpt = AuthenticationService.authenticatedUserFor(request)
+        CheckoutService.processSubscription(userData, idUserOpt).map {
+          case CheckoutResult(_, guestUser: GuestUser) =>
+            Some(FinishAccountForm().bind(guestUser.toFormParams))
+          case CheckoutResult(_, _:MinimalIdUser) =>
+            None
+        }.map(formOpt => Ok(views.html.checkout.thankyou(formOpt)))
       }
     )
   }
 
   def thankyou = GoogleAuthenticatedStaffAction(Ok(views.html.checkout.thankyou()))
```

Comment:
This is currently a mock page (if you look into it, order reference number and subscription code are hardcoded). Happy to handle that as a redirect and pass the form params as query strings if we really have a requirement for bookmarking.